### ISSUE TITLE
Adjust database and crawler to add domain_id to domain_list&articles table

### ISSUE THREAD

**Issue by oliversauter** [WorldBrain/cortex#72]
As discussed in the call, we need to add an ID number to each domain in the 'domain_list' table and replace the 'domain_name' field with 'domain_id' in every article.

@tobisanya mentioned that this should be done when saving articles in the db via the crawler. 

@tobisanya @vpolisky please discuss :)

Also elastic-search needs to be configured to index the ID instead of the name.

**Comment by vpolisky** [comment]
@oliversauter By _domain_list_ you actually mean the _Source_ table?

An _id_ field is implicitly added to any _Source_ when the source is saved into the database, so we don't need to do it explicitly though it doesn't appear in the definition of the model. Or do you mean something else?

Replacing domain name by an id makes, from my view, no sense, since crawler needs domain name to start crawling. How it would be possible to crawl a domain say with id=10? Where should the crawler start the search? Or do I misunderstand the problem?

> saving articles in the db via the crawler

The spider doesn't save articles, but only the URL's found. Saving articles is the job of the content extractor, isn't it? Anyway, an id field is implicitly created by Django for each article.

**Comment by tobisanya** [comment]
@vpolisky @oliversauter Sorry for the confusion. I meant Content Extractor not Crawler which is why i mentioned in the call that @arpitgogia worked on it... Sorry again for the confusion

**Comment by oliversauter** [comment]
ah ok! makes sense. As @vpolisky already mentioned, with saving the domains, we already give each domain a primary key, which is the same as domain_id, right?

This Id will then be stored by the 'content_extractor' when a new article is parsed and stored in the 'articles' table. So this has to be adapted for us being able to filter by domain in the search results.

**Comment by tobisanya** [comment]
> ah ok! makes sense. As @vpolisky already mentioned, with saving the domains, we already give each domain a primary key, which is the same as domain_id, right?

Yes

**Comment by tobisanya** [comment]
> This Id will then be stored by the 'content_extractor' when a new article is parsed and stored in the 'articles' table. So this has to be adapted for us being able to filter by domain in the search results.

Yes. Once the `domain_id` is stored in the `articles` table, it is automatically synced to `ES`. So searching by `domain_id` will automatically work :)
So the only task is to store the `domain_id` in the `articles` table

**Comment by vpolisky** [comment]
Ah ok, so, what was really meant, was to replace _domain_name_ by _domain_id_ in the _Article_ table. I think, from design view, it would be cleaner to have a foreign key on _AllUrl_, since in _AllUrl_ we already have a foreign key on _Source_. Otherwise we store same data on multiple places: _Source_, _AllUrl_, _Article_. So, I would propose simply to replace _url = URLField_ by _url = ForeignKey(AllUrl)_ in the Article table. Does it make sense? Would it lead to problems with _ES_?

**Comment by tobisanya** [comment]
@vpolisky it makes sense. As long as its the same id from `AllUrl` that is being used with search.

**Comment by oliversauter** [comment]
ok cool, seems like we have a resolution. Its important that you get this rolling, because it blocks https://github.com/WorldBrain/webmarks/issues/29 from being completely finished.
